ide/nativeimage.api/src/org/netbeans/modules/nativeimage/api/SourceInfo.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.netbeans.modules.nativeimage.api;
+
+/**
+ * A source file information in the debuggee program.
+ *
+ * @since 0.2
+ */
+public final class SourceInfo {
+
+    private final String fileName;
+    private final String fullName;
+
+    private SourceInfo(String fileName, String fullName) {
+        this.fileName = fileName;
+        this.fullName = fullName;
+    }
+
+    /**
+     * Get the file name, or <code>null</code> when unknown. May return a relative path.
+     *
+     * @since 0.2
+     */
+    public String getFileName() {
+        return fileName;
+    }
+
+    /**
+     * Get the full file name, or <code>null</code> when unknown. Returns an absolute path, if any.
+     *
+     * @since 0.2
+     */
+    public String getFullName() {
+        return fullName;
+    }
+
+    /**
+     * Creates a builder to build a new {@link SourceInfo}.
+     *
+     * @since 0.2
+     */
+    public static Builder newBuilder() {
+        return new Builder();
+    }
+
+    @Override
+    public String toString() {
+        return "SourceInfo{" + "fileName=" + fileName + ", fullName=" + fullName + '}';
+    }
+
```

Comment:
If `SourceInfo` is used in `Map`, then it needs proper `equals` and `hashCode`.

Reply:
Yes, done.